### ISSUE TITLE
TooManyBitmapsException cause no display AGAIN

### ISSUE THREAD

**Issue by narychen** [facebook/fresco#1351]
This problem has been issued many times, but I don't think it has been solved.
I added a comment at the bottom of this issue https://github.com/facebook/fresco/issues/213
The image will not show after a long time running if you have a lot of pictures to render because of TooManyBitmapsException.
But imagePipeline.clearCaches() or shutDown() doesn't solve it.
I checked the code in com.facebook.imagepipeline.memory.BitmapCounter and didn't find any code clear the mCount and mSize to zero when you call clearCaches(). 
So I think there may be something missed here.

**Comment by huangjunkun** [comment]
@narychen   Truth is that You are negligent. Look the follow code.
```

  /**
   * Pins the bitmap
   */
  public CloseableReference<Bitmap> pinBitmap(Bitmap bitmap) {
    try {
      // Real decoding happens here - if the image was corrupted, this will throw an exception
      Bitmaps.pinBitmap(bitmap);
    } catch (Exception e) {
      bitmap.recycle();
      throw Throwables.propagate(e);
    }
    if (!mUnpooledBitmapsCounter.increase(bitmap)) {
      bitmap.recycle();
      throw new TooManyBitmapsException();
    }
    return CloseableReference.of(bitmap, mUnpooledBitmapsCounter.getReleaser());
  }
```

**Decrease the mCount and mSize By mUnpooledBitmapsCounter.getReleaser(), CloseableReference<Bitmap>.**

**Comment by narychen** [comment]
@huangjunkun Maybe you were right.
But I added the following code
```
    ImagePipeline imagePipeline = Fresco.getImagePipeline();
    imagePipeline.clearCaches();
    int count = BitmapCounterProvider.get().getCount();
    long size = BitmapCounterProvider.get().getSize();
    FLog.d("LoggingListener", "-"+count+"-"+size);
```
And the output is
D/unknown:LoggingListener: -9-8905536
I execute it a few times, the count is always 9

**Comment by narychen** [comment]
And I set breakpoint at pinBitmap() and call clearCaches(), it will not be called.
![335fb134-1f57-41fe-a756-6a162e945d88](https://cloud.githubusercontent.com/assets/1779816/16763036/b80a6cb0-4859-11e6-845a-cb04df96fb7b.png)

After using a while the mCount raise up to 64 and call clearCaches() still could not decrease it.

![1ed0857d-36f6-471c-94d3-4761cbad6292](https://cloud.githubusercontent.com/assets/1779816/16763098/f0544d02-4859-11e6-9dd6-1a730351fd9f.png)

There must be some leaks at somewhere.

**Comment by huangjunkun** [comment]
@narychen 
Yeah, it must be some leaks at somewhere. You check your code again carefully.

**Comment by narychen** [comment]
I found that quit the app and launch again will lead to bitmapCounter start counting on the number of what you last quit and never decrease down below that number. 
I've got more specific description on https://github.com/facebook/react-native/issues/8677#issuecomment-232281518

I think there maybe some problem with the special region of Android memory where decompressed image stored on On Android 4.x and lower.
Because I noticed that the memory usage is very low at about 20M-30M on my phone of android 4.3. But when I browse a lot of images, the app will crash with the error of
<sharedmem_gpumem_alloc_id:1427>: sharedmem_gpumem_alloc: mmap failed errno 12 Out of memory
